kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,45 @@
+package com.kodilla.patterns.prototype.library;
+
+import com.kodilla.patterns.prototype.Prototype;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype<Library> {
+
+    String name;
+    Set<Book> books = new HashSet<>();
+
+    public Library(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException
+    {
+        return super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException
+    {
+        Library cloneLibrary = super.clone();
+        cloneLibrary.books = new HashSet<>();
```

Comment:
tutaj odwolanie powinno byc przez setter. Sprawdz tez czy to jest konieczne aby wrzucac tam nowy HashSet